locust/stats.py

```diff
@@ -497,7 +497,13 @@ def percentile(self, elapsed_time=0, get_current=False, csv_format=False):
         for perc, _desc, fmt in percentiles_to_report:
             values.append(percentile_helper(perc))
             formatter.append(fmt+'d')
-        
+        '''
+        For the case where there is a None for the calculated percentiles,
+        we skip logging the metric instead of failing hard, so that other
+        metrics will still be written to the log
+        '''
+        if None in values:
```

Comment:
did you ever find cases where not all values were None? or is it an all or nothing?

Reply:
It seems to be all or nothing. I can do some more logging to check though. But if there's no data for the 50%, then there won't be for the 99th. Although the first 3 values usually are populated since they are the label, time and count

Reply:
👍